packages/runtime-common/module-syntax.ts

```diff
@@ -434,12 +440,12 @@ function makeNewField({
     suggestedCardName(fieldRef, fieldDefinitionType),
   );
 
-  if (
-    fieldRef.module.startsWith(baseRealm.url) &&
-    fieldRef.name === 'default'
-  ) {
-    // primitive fields
-    return `@${fieldDecorator.name} ${fieldName} = ${fieldTypeIdentifier.name}(${fieldCardIdentifier.name});`;
+  if (sourceCodeForComputedField) {
+    return `@${fieldDecorator.name} ${fieldName} = ${fieldTypeIdentifier.name}(${fieldCardIdentifier.name}, {
+              computeVia: function () {
+                return ${sourceCodeForComputedField}
```

Comment:
This approach precludes being able to write a multi-line computed function. Perhaps instead, the end developer should be required to pass code that evaluates to a function, and then that function should we used as the computeVia.